test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests.cs

```diff
@@ -1587,6 +1588,118 @@ public async Task RestoreCommand_CentralVersion_ErrorWhenCentralPackageVersionFi
+                {
+                    new PackageSource(pathContext.PackageSource),
+                    new PackageSource("https://feed1"),
+                    new PackageSource("https://feed2"),
+                };
+
+                var projectName = "TestProject";
+                var projectPath = Path.Combine(pathContext.SolutionRoot, projectName);
+                var outputPath = Path.Combine(projectPath, "obj");
+                var dependencyBar = new LibraryDependency(
+                    new LibraryRange(
+                        "foo",
+                        null,
+                        LibraryDependencyTarget.All),
+                    LibraryIncludeFlags.All,
+                    LibraryIncludeFlags.All,
+                    new List<NuGetLogCode>(),
+                    autoReferenced: false,
+                    generatePathProperty: false,
+                    versionCentrallyManaged: false,
+                    LibraryDependencyReferenceType.Direct,
+                    aliases: null,
+                    versionOverride: null);
+
+                var centralVersionFoo = new CentralPackageVersion("foo", VersionRange.Parse("1.0.0"));
+
+                var tfi = CreateTargetFrameworkInformation(new List<LibraryDependency>() { dependencyBar }, new List<CentralPackageVersion>() { centralVersionFoo });
+                var packageSpec = new PackageSpec(new List<TargetFrameworkInformation>() { tfi })
+                {
+                    FilePath = projectPath,
+                    Name = projectName,
+                    RestoreMetadata = new ProjectRestoreMetadata()
+                    {
+                        ProjectName = projectName,
+                        ProjectUniqueName = projectName,
+                        CentralPackageVersionsEnabled = true,
+                        ProjectStyle = ProjectStyle.PackageReference,
+                        OutputPath = outputPath,
+                        Sources = packageSources
+                    }
+                };
+
+                var logger = new TestLogger();
+
+                PackageSourceMapping packageSourceMappingConfiguration = PackageSourceMapping.GetPackageSourceMapping(NullSettings.Instance);
+
+                if (enablePackageSourceMapping)
+                {
+                    string testConfig = Path.Combine(pathContext.WorkingDirectory, "test.config");
+
+                    SettingsTestUtils.CreateConfigurationFile(testConfig, $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSourceMapping>
+        <clear />
+        <packageSource key=""{pathContext.PackageSource}"">
+            <package pattern=""foo"" />
+        </packageSource>
+        <packageSource key=""https://feed1"">
+            <package pattern=""bar"" />
+        </packageSource>
+        <packageSource key=""https://feed2"">
+            <package pattern=""baz"" />
+        </packageSource>
+    </packageSourceMapping>
```

Comment:
Is having more than 1 source is bad practice with central package management?
Still trying to understand why it's the issue.

Reply:
We want to make NuGet secure by default and in this case since the customer is using CPM, they should at least be told they can use package source mapping to be more secure.